test/createEpicMiddleware-spec.js

```diff
@@ -414,4 +414,40 @@ describe('createEpicMiddleware', () => {
 
     expect(epic.called).to.equal(true);
   });
+
+  it('should not allow interference from the public queueScheduler singleton', (done) => {
+    const reducer = (state = [], action) => state.concat(action);
+    const epic1 = action$ =>
+      action$.pipe(
+        ofType('ACTION_1'),
+        mergeMap(() =>
+          of({ type: 'ACTION_2' }, { type: 'ACTION_3' })
+        )
+      );
+
+    const epic2 = action$ =>
+      action$.pipe(
+        ofType('ACTION_2'),
+        map(() => ({ type: 'ACTION_4' }))
+      );
+
+    const rootEpic = combineEpics(epic1, epic2);
+
+    queueScheduler.schedule(() => {
+      const middleware = createEpicMiddleware();
+      const store = createStore(reducer, applyMiddleware(middleware));
+      middleware.run(rootEpic);
+      store.dispatch({ type: 'ACTION_1' });
+
+      expect(store.getState()).to.deep.equal([
+        initAction,
+        { type: 'ACTION_1' },
+        { type: 'ACTION_2' },
+        { type: 'ACTION_3' },
+        { type: 'ACTION_4' }
+      ]);
+
+      done();
```

Comment:
This isn't really "async" in the sense of needing this `done()` but I prefer to have it regardless in case of later refactors later that might cause the test to pass without actually testing any code.